python/lsst/ctrl/mpexec/singleQuantumExecutor.py

```diff
@@ -327,7 +327,17 @@ def findOutputs(collections):
                 if self.clobberOutputs:
                     # only prune
                     _LOG.info("Removing partial outputs for task %s: %s", taskDef, existingRefs)
-                    butler.pruneDatasets(existingRefs, disassociate=True, unstore=True, purge=True)
+                    # Do not purge registry records if this looks like
+                    # an execution butler. This ensures that the UUID
+                    # of the dataset doesn't change.
+                    if butler._allow_put_of_predefined_dataset:
```

Comment:
Use of `_` suggests that maybe a more explicit API to determine execution butler may be warranted?

Reply:
I agree with K-T, if we start depending on execution butler behavior outside daf_butler maybe it's time to add public method/property?

Reply:
Yes, that's the problem, but on the other hand ctrl_mpexec is not a million miles away from having to worry about the internals. I definitely don't want this to be a user visible property.

Reply:
@TallJimbo is hoping that execution butler is a transient concept and will disappear next year.